pkg/console/operator/operator.go

```diff
@@ -135,84 +133,52 @@ func (c *consoleOperator) Key() (metav1.Object, error) {
 	return operatorConfig, err
 }
 
-// sync() is the handler() function equivalent from the sdk
-// this is the big switch statement.
-// TODO: clean this up a bit, its messy
 func (c *consoleOperator) Sync(obj metav1.Object) error {
 	startTime := time.Now()
 	logrus.Infof("started syncing operator %q (%v)", obj.GetName(), startTime)
 	defer logrus.Infof("finished syncing operator %q (%v) \n\n", obj.GetName(), time.Since(startTime))
 
-	operatorConfig := obj.(*consolev1alpha1.Console)
+	operatorConfig := obj.(*consolev1.Console)
```

Comment:
Pretty sure if you swap `Sync()` func here for:


```golang

func (c *consoleOperator) Sync(obj metav1.Object) error {
	startTime := time.Now()
	logrus.Infof("started syncing operator %q (%v)", obj.GetName(), startTime)
	defer logrus.Infof("finished syncing operator %q (%v) \n\n", obj.GetName(), time.Since(startTime))

	operatorConfig := obj.(*consolev1.Console)

	if err := c.handleSync(operatorConfig); err != nil {
		return err
	}
	return nil
}

func (c *consoleOperator) handleSync(config *consolev1.Console) error {

	switch config.Spec.ManagementState {
	case operatorv1.Managed:
		logrus.Println("console is in a managed state.")
		// handled below
	case operatorv1.Unmanaged:
		logrus.Println("console is in an unmanaged state.")
		return nil
		// take a look @ https://github.com/openshift/service-serving-cert-signer/blob/master/pkg/operator/operator.go#L86
	case operatorv1.Removed:
		logrus.Println("console has been removed.")
		return c.deleteAllResources(config)
	default:
		// TODO should update status
		return fmt.Errorf("unknown state: %v", config.Spec.ManagementState)
	}

	// do we need the if(configChanged) update bits?
	outConfig, configChanged, err := sync_v400(c, config)
	if err != nil {
		return err
	}
	if configChanged {
		// TODO: this should do better apply logic or similar, maybe use SetStatusFromAvailability
		_, err = c.operatorClient.Update(outConfig)
		if err != nil {
			return err
		}
	}

	return nil
}

```
It will eliminate the fields that we no longer care about.

Reply:
See here, a previous pass at moving to the `v1`, before things changed: https://github.com/openshift/console-operator/pull/103/files#diff-69ecbb024dab972c8729678abdf9c0faR132

Reply:
It drops the second switch statement entirely.  We dont need it yet.